app/models/post.rb

```diff
@@ -48,16 +25,18 @@ def tags_string=(value)
     @tags_list = @tags_list.uniq
   end
 
-  def create_tags_from_tag_string
-    tags.clear
-    @tags_list.each do |tag_title|
-      existing_tag = Tag.find_by(title: tag_title)
+  def external_post?
+    type == "ExternalPost"
+  end
+
+  def internal_post?
+    type == "InternalPost"
+  end
 
-      if existing_tag
-        tags << existing_tag
-      else
-        tags.create!(title: tag_title, user_id: user_id)
-      end
+  def question_tag
+    if external_post?
+      related_tag = Tag.find_or_create_by(title: "question")
```

Comment:
Not all external posts will be questions. Only the StackExchange posts. Please move this code into the StackExchange Sidekiq worker.